config/config_mgmt.py

```diff
@@ -430,65 +478,84 @@ def mergeItems(it1, it2):
         return D1
 
     """
-    Create a defConfig for given Ports from Default Config File.
+    search Relevant Keys in Config using DFS, This function is mainly
+    used to search port related config in Default ConfigDbJson file.
+    In: Config to be searched
+    skeys: Keys to be searched in In Config i.e. search Keys.
+    Out: Contains the search result
     """
-    def getDefaultConfig(self, ports=list()):
+    def searchKeysInConfig(self, In, Out, skeys):
+
+        found = False
+        if isinstance(In, dict):
+            for key in In.keys():
+                #print("key:" + key)
+                for skey in skeys:
+                    # pattern is very specific to current primary keys in
+                    # config DB, may need to be updated later.
+                    pattern = '^' + skey + '\|' + '|' + skey + '$' + \
+                        '|' + '^' + skey + '$'
+                    #print(pattern)
+                    reg = re.compile(pattern)
+                    #print(reg)
+                    if reg.search(key):
+                        # In primary key, only 1 match can be found, so return
+                        # print("Added key:" + key)
+                        Out[key] = In[key]
+                        found = True
+                        break
+                # Put the key in Out by default, if not added already.
+                # Remove later, if subelements does not contain any port.
+                if Out.get(key) is None:
```

Comment:
if Out.get(key) is None, doesn't it mean we did NOT find key in any skeys? If so, what are we doing here by recursively calling the function? Could you please explain.

Reply:
Here In Dict will be somethign like below:
"LOOPBACK_INTERFACE": {
			"Loopback0": {},
			"Loopback0|2a04:5555:40:4::4e9/128": {
				"scope": "global",
				"family": "IPv6"
			},
			"Loopback0|10.184.8.233/32": {
				"scope": "global",
				"family": "IPv4"
			}
		},
		"CRM": {
			"Config": {
				"polling_interval": "0"
			}
		}
"DEVICE_NEIGHBOR": {
			"Ethernet112": {
				"name": "dccsw01.nw",
				"port": "Eth18"
			},
			"Ethernet114": {
				"name": "dccsw02.nw",
				"port": "Eth18"
			},
			"Ethernet116": {
				"name": "dccsw03.nw",
				"port": "Eth18"
			},
			"Ethernet118": {
				"name": "dccsw04.nw",
				"port": "Eth18"
			}
		},

```
and Out dict must be 
```
DEVICE_NEIGHBOR": {
			"Ethernet114": {
				"name": "dccsw02.nw",
				"port": "Eth18"
			}
}
```
if skeys is [Ethermet114].